otter/test/test_cloud_client.py

```diff
@@ -427,12 +429,56 @@ def _test_throttle(self, cfg_name, stype, method):
         clock.advance(500)
         self.assertEqual(self.successResultOf(result2), 'bar2')
 
+    def _test_tenant(self, cfg_name, stype, method):
+        """
+        Test a specific throttling configuration, and ensure that locks are
+        per-tenant.
+        """
+        locks = WeakLocks()
+        set_config_data(
+            {'cloud_client': {'throttling': {cfg_name: 500}}})
+        self.addCleanup(set_config_data, {})
+        clock = Clock()
+        bracket1 = _default_throttler(locks, clock, stype, method, 'tenant1')
+        if bracket1 is None:
+            self.fail("No throttler for %s and %s" % (stype, method))
+        result1 = bracket1(lambda: 'bar1')
+        bracket2 = _default_throttler(locks, clock, stype, method, 'tenant2')
+        result2 = bracket2(lambda: 'bar2')
+        self.assertNoResult(result1)
+        self.assertNoResult(result2)
+        clock.advance(500)
+        self.assertEqual(self.successResultOf(result1), 'bar1')
+        self.assertEqual(self.successResultOf(result2), 'bar2')
```

Comment:
It creates two throttlers for two different tenants, and then *only* advances time by 500 seconds, and ensures that both requests complete. Since the delay is set to 500, this proves that the delays are independent. If the lock was shared, then after 500 seconds only one of the requests would have gone through, and it would require another 500 seconds for the second one to complete.

But I could be confused about your point, sorry if I'm missing something